### PR TITLE
Asteroid and ship selection enhancements

### PR DESCRIPTION
**Pull request by tatami4**
## Summary
This is a reopening of #4366.
Changes in the game:
1. Player cannot have both ship and asteroid targets simultaneously.
2. Clicking on ship deselects selected asteroid.
3. Clicking on asteroid deselects current ship.
4. Clicking on empty space deselects both ship and asteroid.
5. Asteroid target is now highlighted regardless of distance to flagship.

## Save File
[testsave.txt](https://github.com/endless-sky/endless-sky/files/5860438/Broken.Escorts.2.escorts.stuck.txt)

### CODE REVIEW

File: source/Engine.cpp [hunk 1]
```diff
@@ -1782,6 +1782,10 @@ void Engine::HandleMouseClicks()
 	if(!flagship)
 		return;
 	
+	// Track if player clicked on something.
+	bool clickedAsteroid = false;
+	bool clickedPlanet = false;
+	
 	// Handle escort travel orders sent via the Map.
```
Comment (tehhowch) [inline source/Engine.cpp:1789]: Move these variable declarations closer to the point of first use.

File: source/Engine.cpp [hunk 2]
```diff
@@ -1875,11 +1881,16 @@ void Engine::HandleMouseClicks()
...
 			if(range <= clickRange)
 			{
+				clickedAsteroid = true;
 				clickRange = range;
 				flagship->SetTargetAsteroid(minable);
 			}
 		}
 	}
+	
+	// If the player didn't click on anything, clear ship and asteroid targets.
```
Comment (tehhowch) [inline source/Engine.cpp:1891]:
```suggestion
	// Treat an "empty" click as a request to clear targets.
```

**Review by tehhowch — COMMENTED** [review]
Looks good, just a minor style issue and optional rephrasing of the comment.

**Comment by tatami4** [comment]
#4366 should be closed too.

**Pull request merged**